v-next/hardhat/src/internal/builtin-plugins/solidity/build-system/cache.ts

```diff
@@ -0,0 +1,74 @@
+import path from "node:path";
+
+import {
+  exists,
+  getAccessTime,
+  getAllFilesMatching,
+  getSize,
+  readJsonFile,
+  remove,
+  writeJsonFile,
+} from "@ignored/hardhat-vnext-utils/fs";
+
+export class ObjectCache<T> {
+  readonly #path: string;
+  readonly #defaultMaxAgeMs: number;
+  readonly #defaultMaxSize: number;
+
+  constructor(
+    basePath: string,
+    namespace: string,
+    version: string,
+    defaultMaxAgeMs: number = 7 * 24 * 60 * 60 * 1000, // 1 week
+    defaultMaxSize: number = 2 * 1024 * 1024 * 1024, // 2 GB
```

Comment:
These are the default values we agreed upon [here](https://www.notion.so/nomicfoundation/solidity-compilation-cache-17b578cdeaf580e3ba10c0d907fd6a4d?pvs=4#182578cdeaf580f3aaa6d946581e7ac7). They are not exposed to the user at the moment.